### ISSUE TITLE
Calibration data unavailable after Home Assistant restarts

### ISSUE THREAD

**Issue by EnUfor** [erkexzcx/valetudopng#5]
**Describe the bug**
After Home Assistant restarts (due to update, manual restart, etc.), the `sensor.{{vacuum}}_calibration` entity becomes unavailable. This causes xiaomi-vacuum-map-card to not render as the card's `calibration_source` field now points to an unavailable entity.

Current work-around is to restart the ValetudoPNG docker container which repopulates the missing entity.

**To Reproduce**
1. Run ValetudoPNG and observe calibration and rendered map populate on the MQTT Vacuum
1. Restart or update Home Assistant
1. Observe `_calibration` sensor for vacuum is now Unavailable 
1. Observe xiaomi-vacuum-map-card fails to render due to missing calibration_source

**Expected behavior**
Expected calibration data to retain after home assistant restarts or ValetudoPNG to repopulate/send the calibration data again.

**Screenshots/Outputs**
MQTT entry for Vacuum after Home Assistant has restarted
![image](https://github.com/erkexzcx/valetudopng/assets/50682605/7c9d23da-1efc-4b99-bd6f-51b1b7a0fca1)

Calibration entity
![image](https://github.com/erkexzcx/valetudopng/assets/50682605/f789ae0a-fce9-41b0-b3bc-d8b0500cfab5)

Firefox console
![image](https://github.com/erkexzcx/valetudopng/assets/50682605/092546c0-064e-4ec3-adc6-3d3144561634)

**Container Logs**
```
2023/11/02 00:00:16 Image rendered! drawing:18ms, encoding:56ms, size:36.6kB
2023/11/02 00:00:17 Skipping image render due to min_refresh_int
2023/11/02 00:01:17 Image rendered! drawing:18ms, encoding:57ms, size:36.6kB
2023/11/02 00:01:18 Skipping image render due to min_refresh_int
2023/11/02 00:02:19 Image rendered! drawing:19ms, encoding:56ms, size:36.6kB
2023/11/02 00:02:19 Skipping image render due to min_refresh_int
2023/11/02 00:03:19 Image rendered! drawing:18ms, encoding:55ms, size:36.6kB
2023/11/02 00:04:20 Image rendered! drawing:19ms, encoding:62ms, size:36.6kB
2023/11/02 00:04:21 Skipping image render due to min_refresh_int
etc...
```
Map is still updating correctly, but `_calibration` data is never sent again until the container is restarted

**Comment by mundschenk-at** [comment]
I don't think this is fixed yet, due to the discovery topics not being retained. See #9.

**Comment by EnUfor** [comment]
Never had a chance to reopen this issue with feedbacn. After a while of testing and a few Home Assistant updates/restarts (up to 2024.1.1 currently), I'm still seeing the same issue.

I was able to clear the MQTT messages and re-connected the client vacuums to get the updated sensor names shown below. Is there further steps we can take to diagnose this issue?

![Screenshot_20240108-123359](https://github.com/erkexzcx/valetudopng/assets/50682605/124db914-fb85-4ccc-8f98-40344c144149)

**Comment by mundschenk-at** [comment]
I don't think so, it's the result of the discovery topics not having `retain` set, as described in #9.
